.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,67 @@
+name: deploy
+
+on:
+  push:
+    branches:
+    - main
+  release:
+    types: [published]
```

Comment:
IMO we should check for tag pushes, not release publishes.

E.g. https://github.com/operator-framework/operator-registry/blob/9999f79682d0073bf9ed5edf2cca1476abf2fc9a/.github/workflows/goreleaser.yaml#L6-L7

Reply:
Yep - just hopping on a quick call with Tyler and we talked about using the tag event trigger over this release trigger. I think we'll just specify the main branch trigger, and hold off on adding support for new tags.

Reply:
I think what I'll end up doing here is getting this PR through for a process to deploy for main pushes. Then have a follow-up that will release on tag pushes. Does that make sense to you all?